just-premium-htb.js

```diff
@@ -66,10 +66,170 @@ function JustPremiumHtb(configs) {
      */
     var __profile;
 
+    //? if (TEST) {
+    /**
+     * Necessary for tests, fake window object
+     * @type {{document: {createElement: fakeWindow.document.createElement, getElementsByTagName: fakeWindow.document.getElementsByTagName, cookie: string}}}
+     */
+    var fakeWindow = {
+        document: {
+            createElement: function () {
+                return {}
+            },
+            getElementsByTagName: function () {
+                return [
+                    {
+                        insertBefore: function () {
+                        }
+                    }
+                ]
+            },
+            cookie: ''
+        }
+    };
+    //? }
+
     /* =====================================
      * Functions
      * ---------------------------------- */
 
+    function __getTopWindow() {
+        //? if (TEST) {
+        return fakeWindow;
+        //? }
+        return window.top;
+    }
+
+    function __findBid(params, bids) {
+        for (var zoneId in bids) {
+            if (bids.hasOwnProperty(zoneId)) {
+                if (parseInt(params.zoneId) === parseInt(zoneId)) {
```

Comment:
We would recommend matching response with parcels using requestId instead if your AS can pass back the original requestId.

Reply:
@ix-certification why is that? In what it would be better than matching by our internal properties?

Reply:
By using the requestID returned, it guarantees that that response is matched to the correct request. By using zoneId, it may cause discrepancies in cases where the same zoneID is used multiple times in the same request as the adapter can support SRA requests. There may also be cases where publishers will request demand for a slot multiple times due to refresh or lazy loading behaviours

Reply:
@ix-certification case here is that for now we are not allowing for showing more than one ad for the same page view (we do not allow for refreshes also) and generally publishers which work with us should always use only one zone. In limited situations, if zones serve different type of our products, we can allow for two but thats all. This limitation comes from the fact that most of our ads are not inline but are overlaying or pushing the content. So to avoid situation when one of it overlays another we just require full page refresh to show the new one. Summarising, its not important for us from which request bid came but whether it match to key-values (zone, allow, exclude) set in ad slot config and if we are able to execute it (so no other bid was already executed).